aiohttp/client_reqrep.py

```diff
@@ -1011,36 +1006,33 @@ def get_encoding(self) -> str:
         mimetype = helpers.parse_mimetype(ctype)
 
         encoding = mimetype.parameters.get("charset")
-        if encoding:
-            try:
-                codecs.lookup(encoding)
-            except LookupError:
-                encoding = None
-        if not encoding:
-            if mimetype.type == "application" and (
-                mimetype.subtype == "json" or mimetype.subtype == "rdap"
-            ):
-                # RFC 7159 states that the default encoding is UTF-8.
-                # RFC 7483 defines application/rdap+json
-                encoding = "utf-8"
-            elif self._body is None:
-                raise RuntimeError(
-                    "Cannot guess the encoding of " "a not yet read body"
-                )
-            else:
-                encoding = chardet.detect(self._body)["encoding"]
```

Comment:
Given this is a public, documented method, I think we should call self._fallback_encoding() here, and keep the logic similar to before. That probably means not using the try/except logic in text()..

Reply:
No, the old code only calls it if `not encoding`, after getting it from the header. The only difference between your proposal is that it won't try to decode as utf-8 first, which is not something I'm strongly tied to, I think it could damage performance is some cases and improve it in others.

Reply:
OK, I think I understand what you're asking for. I'll see if I can deliver.

Reply:
@Dreamsorcerer I have the structure of the method largely preserved here. 

Somewhat related: Should `get_encoding()` return the value from the header _or_ should it return Python's value?

For instance, if the header says the encoding is "windows-1252", should this function return "windows-1252" or "cp1252"?

Reply:
Probably whatever it did before, which I guess is Python's? Although both those values appear to be valid in .encode().

Reply:
The previous behavior appeared to return the header value, actually. It calls codecs.lookup(), but it doesn't actually return from it, nor assign the output to any variable. Then flow control continues to the bottom where it will return the header value unchanged

Reply:
Right, I don't think it really matters, as both appear to work. But, I guess the Python version is what I'd expect for the most part.